app/controllers/publishers/base_controller.rb

```diff
@@ -34,16 +34,17 @@ def current_publisher_preferences
     PublisherPreference.find_by(publisher_id: current_publisher.id, school_group_id: current_organisation.id)
   end
 
-  def check_user_last_activity_at
-    return redirect_to logout_endpoint if current_publisher&.last_activity_at.blank?
+  def authenticate_publisher!
```

Comment:
Not sure about the name of this method. are we authenticating the publisher or just checking the last activity?

Reply:
I wasn't sure about the name at first, but actually this method was implicitly authenticating publishers already – in the first line, if `current_publisher` is `nil`, it also redirects to the logout endpoint because of the `&.`. The `#check_session` method only checks organisation keys in the session.

I also changed this to go to the login page if you don't already have a session, because it seems weird to redirect the user to DSI in that case. But maybe it should?